services/core-api/app/api/parties/party_appt/models/mine_party_appt.py

```diff
@@ -64,9 +64,9 @@ def validate_permit():
                     'Permits cannot be associated with non permit-linked contact types.')
 
         def validate_tsf():
-            if self.mine_party_appt_type_code == 'EOR':
+            if self.mine_party_appt_type_code in TSF_ALLOWED_CONTACT_TYPES:
                 if not self.mine_tailings_storage_facility:
-                    raise AssertionError('The associated TSF is required for Engineer of Records.')
+                    raise AssertionError('The associated TSF is required for Engineer of Records or Qualified Persons.')
```

Comment:
Probably not a big deal, but since Qualified Persons can exist for other Mine contacts, maybe we could change the last part to something like 'TSF related qualified persons'?